internal/tiltfile/version/version.go

```diff
@@ -19,26 +27,37 @@ func (e Extension) NewState() interface{} {
 	}
 }
 
-func (Extension) OnStart(env *starkit.Environment) error {
-	return env.AddBuiltin("version_settings", setVersionSettings)
+func (e Extension) OnStart(env *starkit.Environment) error {
+	return env.AddBuiltin("version_settings", e.setVersionSettings)
 }
 
-func setVersionSettings(thread *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
-	var checkUpdates bool
-	if err := starkit.UnpackArgs(thread, fn.Name(), args, kwargs,
-		"check_updates", &checkUpdates); err != nil {
-		return nil, err
-	}
+func (e Extension) setVersionSettings(thread *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var constraint string
 
-	err := starkit.SetState(thread, func(settings model.VersionSettings) model.VersionSettings {
-		if checkUpdates {
-			settings.CheckUpdates = true
-		} else {
-			settings.CheckUpdates = false
+	err := starkit.SetState(thread, func(settings model.VersionSettings) (model.VersionSettings, error) {
+		if err := starkit.UnpackArgs(thread, fn.Name(), args, kwargs,
+			"check_updates?", &settings.CheckUpdates,
+			"constraint?", &constraint,
+		); err != nil {
+			return settings, err
 		}
-		return settings
+		return settings, nil
 	})
 
+	if constraint != "" {
+		ver, err := semver.Parse(e.tiltVersion)
+		if err != nil {
+			return nil, errors.Wrapf(err, "internal error parsing tilt version '%s'", e.tiltVersion)
+		}
+		rng, err := semver.ParseRange(constraint)
+		if err != nil {
+			return nil, errors.Wrapf(err, "error parsing version constraint")
+		}
+		if !rng(ver) {
+			return nil, fmt.Errorf("you are running Tilt version %s, which doesn't match the version constraint specified in your Tiltfile: '%s'", e.tiltVersion, constraint)
```

Comment:
It would be nice to include a helpful error messaging like linking to the website saying howto upgrade.

Also explaining which version they should be running.

Reply:
> It would be nice to include a helpful error messaging like linking to the website saying howto upgrade.

Good call. I'll need to make sure we a page appropriate to link to.

> Also explaining which version they should be running.

Yeah, that would be nice, but I don't see a cheap way to do it with the information available, and it'd greatly increase the complexity of this feature to figure it out. We're currently logging the constraint. I'm inclined to wait to see how much use this gets / whether that's a problem before worrying about it. Also how this fits into #3125.

Reply:
>Good call. I'll need to make sure we a page appropriate to link to. 
https://docs.tilt.dev/upgrade.html this is what I used into my plugin. https://github.com/windmilleng/tilt-extensions/pull/9 

>Yeah, that would be nice, but I don't see a cheap way to do it with the information available, and it'd greatly increase the complexity of this feature to figure it out. We're currently logging the constraint. 

Ah, I couldn't see a log message for this, so I assumed it wasn't there.

Was just a suggestion I don't think you should complicate it too much.